hedera-node/hedera-schedule-service-impl/src/test/java/com/hedera/node/app/service/schedule/impl/test/handlers/ScheduleDeleteHandlerTest.java

```diff
@@ -0,0 +1,153 @@
+        final var txn = scheduleDeleteTransaction();
+        givenSetupForScheduleDelete(txn);
+        given(schedulesById.get(scheduleID.getScheduleNum())).willReturn(schedule);
+
+        final var scheduleDeleteMeta =
+                subject.preHandle(txn, scheduleDeleter, keyLookup, scheduleStore);
+        assertEquals(scheduleDeleter, scheduleDeleteMeta.payer());
+        assertEquals(List.of(), scheduleDeleteMeta.requiredNonPayerKeys());
+        assertEquals(SCHEDULE_IS_IMMUTABLE, scheduleDeleteMeta.status());
+    }
+
+    @Test
+    // when the delete schedule tx is not signed with admin key of scheduled tx,
+    // then fail with INVALID_SIGNATURE
+    void scheduleDeleteFailWhenNotSignedWithAdminKey() throws DecoderException {
+        final var txn = scheduleDeleteTransaction();
+        givenSetupForScheduleDelete(txn);
+
+        Key key =
+                Key.newBuilder()
+                        .setEd25519(ByteString.copyFromUtf8("aaaaaaaaaaaaaaaaaaaaaabbbbbbbbbb"))
+                        .build();
+        given(schedule.adminKey()).willReturn(Optional.of(JKey.mapKey(key)));
+        given(keyLookup.getKey(scheduleDeleter))
+                .willReturn(KeyOrLookupFailureReason.withKey(adminKey));
+        given(schedulesById.get(scheduleID.getScheduleNum())).willReturn(schedule);
+
+        final var scheduleDeleteMeta =
+                subject.preHandle(txn, scheduleDeleter, keyLookup, scheduleStore);
+        assertEquals(scheduleDeleter, scheduleDeleteMeta.payer());
+        assertEquals(List.of(), scheduleDeleteMeta.requiredNonPayerKeys());
+        assertEquals(INVALID_SIGNATURE, scheduleDeleteMeta.status());
+    }
+
+    @Test
+    void handleNotImplemented() {
+        assertThrows(UnsupportedOperationException.class, () -> subject.handle(metaToHandle));
+    }
+
+    private void givenSetupForScheduleDelete(TransactionBody txn) {
+        scheduledTxn =
+                TransactionBody.newBuilder()
+                        .setTransactionID(
+                                TransactionID.newBuilder().setAccountID(scheduler).build())
+                        .setCryptoCreateAccount(CryptoCreateTransactionBody.getDefaultInstance())
+                        .build();
+        scheduledMeta =
+                new SigTransactionMetadata(
+                        asOrdinary(
+                                txn.getScheduleCreate().getScheduledTransactionBody(),
+                                txn.getTransactionID()),
+                        scheduler,
+                        OK,
+                        schedulerKey,
+                        List.of());
+    }
+
+    private TransactionBody scheduleDeleteTransaction() {
+        return TransactionBody.newBuilder()
+                .setTransactionID(TransactionID.newBuilder().setAccountID(scheduleDeleter))
+                .setScheduleDelete(
+                        ScheduleDeleteTransactionBody.newBuilder().setScheduleID(scheduleID))
```

Comment:
Please also add Parity tests like [this](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractCreateHandlerParityTest.java), to port `SigRequiremenetsTest` unit tests related to scheduleDelete

Reply:
I will do this as a separate PR. Created issue #4706

Reply:
Parity tests implemented